Include/internal/pycore_magic_number.h

```diff
@@ -0,0 +1,287 @@
+#ifndef Py_INTERNAL_MAGIC_NUMBER_H
+#define Py_INTERNAL_MAGIC_NUMBER_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+/*
+
+Magic word to reject .pyc files generated by other Python versions.
+It should change for each incompatible change to the bytecode.
+
+The value of CR and LF is incorporated so if you ever read or write
+a .pyc file in text mode the magic number will be wrong; also, the
+Apple MPW compiler swaps their values, botching string constants.
+
+There were a variety of old schemes for setting the magic number.
+The current working scheme is to increment the previous value by
+10.
+
+Starting with the adoption of PEP 3147 in Python 3.2, every bump in magic
+number also includes a new "magic tag", i.e. a human readable string used
+to represent the magic number in __pycache__ directories.  When you change
+the magic number, you must also set a new unique magic tag.  Generally this
+can be named after the Python major version of the magic number bump, but
+it can really be anything, as long as it's different than anything else
+that's come before.  The tags are included in the following table, starting
+with Python 3.2a0.
```

Comment:
I just moved this comment here from the other file, but I did notice this last sentence seems to be no longer true.  There are two 3.2 magic tags mentioned below but that's it.  I think it's just entirely based on version number now.  Does it make sense to remove this paragraph?